### PR TITLE
Apex-pie config panel and color-change config element added

### PR DESCRIPTION
**Pull request by 100starnight**
## Description
Implementing the config panel for the apex-pie chart, including the color-changer element, legend toggle, and percentages toggle.

##Screenshots
Screen Shot 2020-09-11 at 1 04 57 PM

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/components/navi-visualization-config/apex-pie.js [hunk 1]
```diff
@@ -10,5 +10,22 @@
  * />
  */
 import Component from '@glimmer/component';
+import { set, action } from '@ember/object';
+import { copy } from 'ember-copy';
+import { tagName } from '@ember-decorators/component';
 
-export default class NaviVisualizationConfigApexPieComponent extends Component {}
+@tagName('')
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/apex-pie.js:17]:
```suggestion
```

This does not apply for glimmer components so this and the import can be removed

File: packages/core/addon/components/navi-visualization-config/apex-pie.js [hunk 2]
```diff
@@ -10,5 +10,22 @@
...
+  /**
+   * Method to update the config of the chart
+   * @method updateLegendVisible
+   * @param {string} type - the field of the config being updated
+   * @param {boolean} value - the new value of the field being updated
+   */
+  @action
+  updateChart(type, value) {
+    const newOptions = copy(this.args.options);
+    set(newOptions, 'series.config.' + type, value);
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/apex-pie.js:28]:
```suggestion
    set(newOptions, `series.config.${type}`, value);
```

File: packages/core/addon/components/navi-visualization-config/apex-pie.js [hunk 3]
```diff
@@ -10,5 +10,22 @@
...
+import { set, action } from '@ember/object';
+import { copy } from 'ember-copy';
+import { tagName } from '@ember-decorators/component';
 
-export default class NaviVisualizationConfigApexPieComponent extends Component {}
+@tagName('')
+export default class NaviVisualizationConfigApexPieComponent extends Component {
+  /**
+   * Method to update the config of the chart
+   * @method updateLegendVisible
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/apex-pie.js:21]:
```suggestion
   * @method updateChart
```

File: packages/core/addon/components/navi-visualization-config/color-change.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { tagName } from '@ember-decorators/component';
+import { getRequestDimensions } from 'navi-core/utils/chart-data';
+import { assignColors } from 'navi-core/utils/enums/denali-colors';
+
+@tagName('')
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/color-change.js:21]:
```suggestion
```

File: packages/core/addon/utils/enums/denali-colors.js [hunk 5]
```diff
@@ -41,16 +41,20 @@ export function fetchColor(index, type = 'graph') {
 }
 
 export function assignColors(labels, configColors) {
+  // console.log('labels:\n', labels);
+  // console.log('configColors:\n', configColors);
```
Comment (kevinhinterlong) [inline packages/core/addon/utils/enums/denali-colors.js:45]: Some cleanup to do here

File: packages/core/addon/components/navi-visualization-config/color-change.hbs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        </div>
+     </div>
+    </PowerSelect>
+  </div>
+  <div class='color-change-config__color-select'>
+    <span class='color-change-config__color-select__label'>
+      Color:
+    </span>
+    <input
+      onchange={{this.updateChartColor}}
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/color-change.hbs:34]:
```suggestion
      {{on "change" this.updateChartColor}}
```

File: packages/core/addon/components/navi-visualization-config/color-change.hbs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      class='color-change-config__color-select__input'
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/color-change.hbs:35]: `'` -> `"` in templates

**Review by jkusa — APPROVED** [review]

**Pull request merged**